src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/ElasticsearchCaseSearchOperation.java

```diff
@@ -110,12 +116,56 @@ private CaseSearchResult toCaseDetailsSearchResult(MultiSearchResult multiSearch
                 throw new BadSearchRequest(errMsg);
             }
             if (response.searchResult != null) {
+                buildCaseTypesResults(response, caseTypeResults, crossCaseTypeSearchRequest);
                 caseDetails.addAll(searchResultToCaseList(response.searchResult));
                 totalHits += response.searchResult.getTotal();
             }
         }
 
-        return new CaseSearchResult(totalHits, caseDetails);
+        return new CaseSearchResult(totalHits,caseDetails,caseTypeResults);
+    }
+
+    private void buildCaseTypesResults(
+        MultiSearchResult.MultiSearchResponse response,
+        List<CaseTypeResults> caseTypeResults,
+        CrossCaseTypeSearchRequest crossCaseTypeSearchRequest) {
+        if (hitsIsNotEmpty(response)) {
+            String indexName = getIndexName(response);
+            caseTypeResults.add(new CaseTypeResults(getCaseTypeIDFromIndex(indexName,
+                crossCaseTypeSearchRequest.getCaseTypeIds()),
+                response.searchResult.getTotal())
+            );
+        }
+    }
+
+    private String getIndexName(MultiSearchResult.MultiSearchResponse response) {
+        String quotedIndexName =  response.searchResult.getJsonObject().getAsJsonObject(HITS).get(HITS)
+            .getAsJsonArray().get(0).getAsJsonObject().get("_index").toString();
+        String unquotedIndexName = quotedIndexName.replaceAll("\"", "");
+        return unquotedIndexName;
+    }
+
+    private boolean hitsIsNotEmpty(MultiSearchResult.MultiSearchResponse response) {
+        return response.searchResult.getJsonObject().getAsJsonObject(HITS).get(HITS).getAsJsonArray().size() != 0;
+    }
+
+    private String getCaseTypeIDFromIndex(final String index, List<String> caseTypeIds) {
+        String caseTypeIdGroupRegex = applicationParams.getCasesIndexNameCaseTypeIdGroup();
+        int caseTypeIdGroupPosition = applicationParams.getCasesIndexNameCaseTypeIdGroupPosition();
+        Pattern pattern = Pattern.compile(caseTypeIdGroupRegex);
+        Matcher m = pattern.matcher(index);
+        if (m.matches() && m.groupCount() > 1) {
+            return caseTypeIds.stream().filter(
+                caseTypeId -> caseTypeId.equalsIgnoreCase(m.group(caseTypeIdGroupPosition))
+            ).findFirst().orElseThrow(() -> {
+                log.error("Cannot match any known case type id from index '{}' extracted case type id : {}", index, m.group(1));
+                throw new ServiceException("Cannot determine case type id from ES index name - unknown extracted case type id");
+            });
+        } else {
+            log.error("Cannot determine case type id from index name: '{}'. No capturing group configured or capturing group not matching: '{}'.",
+                index, caseTypeIdGroupRegex);
+            throw new ServiceException("Cannot determine case type id from ES index name - cannot extract case type id");
+        }
     }
```

Comment:
@danlysiak improved the logic to extract the case type ID. Now is basically not in the code, but it's in the 2 new config properties I've introduced.
Let me know if it's better now

Reply:
Looks good! Just a couple of small comments further below.

Reply:
done